flink-streaming-java/src/main/java/org/apache/flink/streaming/api/datastream/DataStream.java

```diff
@@ -526,8 +526,14 @@ public DataStream<T> global() {
      * in the set time, the stream terminates.
      *
      * @return The iterative data stream created.
+     * @deprecated This method is deprecated since Flink 1.19. The users are recommended to use
+     *     Iteration API in Flink ML instead.
```

Comment:
We should not recommend users to use Iteration API in Flink ML instead. It doesn't make sense that a user who doesn't need any ML algorithm would have to use Flink ML only to get access to the Iteration API.

I'd suggest to the following:
> The only known use case of this Iteration API comes from Flink ML, which already has its own implementation of iteration and  no longer uses this API. If there's any use cases other than Flink ML that needs iteration support, please reach out to dev@flink.apache.org and we can consider making the Flink ML iteration implementation a separate common library.